beacon-chain/rpc/prysm/v1alpha1/validator/proposer_bellatrix_test.go

```diff
@@ -390,20 +266,25 @@ func TestServer_setExecutionData(t *testing.T) {
 		require.LogsContain(t, hook, "builderGweiValue=1 localBoostPercentage=1 localGweiValue=1")
 	})
 	t.Run("Builder configured. Builder returns fault. Use local block", func(t *testing.T) {
+		bb := builderTest.DefaultBuilderService(t, version.Capella, true)
+		bb.ErrGetHeader = errors.New("fault")
+		bb.Cfg = &builderTest.Config{BeaconDB: beaconDB}
+		vs.BlockBuilder = bb
+
 		blk, err := blocks.NewSignedBeaconBlock(util.NewBeaconBlockCapella())
 		require.NoError(t, err)
-		vs.BlockBuilder = &builderTest.MockBuilderService{
-			ErrGetHeader:  errors.New("fault"),
-			HasConfigured: true,
-			Cfg:           &builderTest.Config{BeaconDB: beaconDB},
-		}
+		require.NoError(t, vs.BeaconDB.SaveRegistrationsByValidatorIDs(ctx, []primitives.ValidatorIndex{blk.Block().ProposerIndex()},
+			[]*ethpb.ValidatorRegistrationV1{{FeeRecipient: make([]byte, fieldparams.FeeRecipientLength), Timestamp: uint64(util.DefaultGenesisTime.Unix()), Pubkey: make([]byte, fieldparams.BLSPubkeyLength)}}))
 		vs.ExecutionEngineCaller = &powtesting.EngineClient{PayloadIDBytes: id, ExecutionPayloadCapella: &v1.ExecutionPayloadCapella{BlockNumber: 4}, BlockValue: 0}
+		chain := &blockchainTest.ChainService{Genesis: util.DefaultGenesisTime, Block: blk}
+		vs.TimeFetcher = chain
+		vs.HeadFetcher = chain
+
 		b := blk.Block()
 		localPayload, _, err := vs.getLocalPayload(ctx, b, capellaTransitionState)
 		require.NoError(t, err)
 		builderPayload, builderKzgCommitments, err := vs.getBuilderPayloadAndBlobs(ctx, b.Slot(), b.ProposerIndex())
-		require.ErrorIs(t, consensus_types.ErrNilObjectWrapped, err) // Builder returns fault. Use local block
-		require.DeepEqual(t, [][]uint8(nil), builderKzgCommitments)
+		require.ErrorContains(t, "fault", err) // Builder returns fault. Use local block
```

Comment:
Test setup was wrong - the `ErrGetHeader` field was set, but this error was never returned because the builder failed earlier.